### PR TITLE
Add support for crazy low level TLS-MAC unpacking

### LINKED ISSUE #629: AES SKEY CIPHER support requires support of TLS_VERSION, TLS_MAC and TLS_MCA_SIZE options
When AES SKEYMGMT is supported and all operations are forced on the token, OpenSSL expects the cipher provider to support special TLS options to deal with TLS records at the cipher level.

This is a gross layering violation that greatly complicates the code, unfortunately, like for PKCS1.5 this layering violation needs to be done to prevent side-channel attacks on the de-padding of the records.

### PR DESCRIPTION
**Pull request by simo5**
#### Description

OpenSSL embedded some TLS specific de-padding and MAC handling into the actual ciphers ... this is crazy, but whatever.

This commit implements manual de-padding and then MAC extraction in constant time.

I believe this is used in rare case for MAC-then-Encrypt cases and it is normally not exercised much, but TLS-Fuzzer does exercise these corner cases and was failing tests when using latest OpenSSL with CIPHER EVP_SKEY enable in pkcs11-provider.

Fixes #629

#### Checklist

- [x] Code modified for feature
- [x] Test suite already covers with functionality tests
- ~[ ] Documentation updated~

#### Reviewer's checklist:

- [ ] Any issues marked for closing are addressed
- [ ] There is a test suite reasonably covering new functionality or modifications
- [ ] This feature/change has adequate documentation added
- [ ] Code conform to coding style that today cannot yet be enforced via the check style test
- [ ] Commits have short titles and sensible commit messages
- [ ] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

File: src/cipher.c [hunk 1]
```diff
@@ -447,32 +466,210 @@ static int p11prov_cipher_decrypt_skey_init(void *ctx, void *keydata,
...
     }
+    session_handle = p11prov_session_handle(cctx->session);
 
     switch (cctx->operation) {
     case CKF_ENCRYPT:
-        rv = p11prov_EncryptUpdate(cctx->provctx,
-                                   p11prov_session_handle(cctx->session),
-                                   (void *)in, inlen, out, &outlen);
+        if (cctx->tlsver != 0) {
+            /* Custom handle padding */
```
Comment (Jakuje) [inline src/cipher.c]: ?
```suggestion
            /* Custom padding handling */
```
or
```suggestion
            /* Handle custom padding */
```

Comment (simo5) [reply src/cipher.c]: Same difference, what about I remove the comment entirely given it is obvious we are handling padding on our own?

Comment (Jakuje) [reply src/cipher.c]: Fine for me. Just the wording sounded odd.

File: src/cipher.c [hunk 2]
```diff
@@ -447,32 +466,209 @@ static int p11prov_cipher_decrypt_skey_init(void *ctx, void *keydata,
...
+                rv = CKR_ARGUMENTS_BAD;
+                P11PROV_raise(cctx->provctx, rv, "Invalid input buffer size");
+                return RET_OSSL_ERR;
+            }
+            /* add the padding, relies on in == out and therefore enough
+             * space available in the buffer */
+            memset(&out[inl], padval, padsize);
+
+            /* in TLS mode we must use single shot decryption to properly
+             * auto-finalize the session as OpenSSL won't */
```
Comment (Jakuje) [inline src/cipher.c]:
```suggestion
            /* in TLS mode we must use single shot encryption to properly
             * auto-finalize the session as OpenSSL won't */
```

**Review by Jakuje — APPROVED** [review]
